### DISCUSSION TITLE
Parametrization of Parameter of Interest

### DISCUSSION THREAD

**Discussion by GiacomoDaMolin** [scikit-hep/pyhf#2399] [Ideas]
Dear experts,

I would like to suggest to include a way to parametrize the Parameter of Interest of the fit.
This would be very useful in analyses that, for example, use ratios or double ratios of channels to reduce the impact of some systematics and achieve the best possible precision, as I am personally trying to do in a CMS analysis. Other cases that I believe would be benefiting from such an implementation would be the ones where the aim is not to measure a cross section, but some polarization/helicity or particle correlations.

Please let me know if such a feature is planned to be implemented.
Kind regards,
Giacomo

**Comment by alexander-held** [comment]
Hi Giacomo, please see #850 for this feature request and #1991 for an experimental implementation.

Ratios of parameters are a special case which you can implement without the need for custom expressions by using a suitable choice of normalization factors. If you want to measure the ratio of normalizations for sample A and sample B, then define a normalization factor `r` (the ratio) and attach it to sample A. Define `b` attached to both samples A and B. You then effectively scale sample A by `a=r*b`, which is another way of saying that you have defined `r:=a/b` (but without needing to go through a new feature, and your workspace does not contain a normalization factor `a` explicitly). This only works in this special case because the expression can be written as a product of normalization factors, which corresponds to attaching multiple normalization factors to the same sample (they compose multiplicatively).
